modules/core/04-channel/v2/keeper/msg_server.go

```diff
@@ -92,18 +101,66 @@ func (k *Keeper) RecvPacket(ctx context.Context, msg *channeltypesv2.MsgRecvPack
 		return nil, errorsmod.Wrap(err, "invalid address for msg Signer")
 	}
 
-	_ = signer
+	switch err {
+	case nil:
```

Comment:
Just noticed one thing: this case can never be true, since we check for `err != nil` above. I believe this is supposed to be a different `err` (the one at line 92? But in that case we still don't need this `case`)

Reply:
really nice catch! Also highlights the missing NoOp case, I'll push a fix for this